cypress/components/flat-table/flat-table.cy.tsx

```diff
@@ -0,0 +1,2595 @@
+          assertCssValueIsApproximately($el, "height", rowHeight);
+        });
+
+        for (let i = 0; i < 4; i++) {
+          flatTableHeaderCells()
+            .eq(i)
+            .find("div")
+            .eq(0)
+            .should("have.css", "padding-left", padding)
+            .and("have.css", "padding-right", padding)
+            .and("have.css", "font-size", fontSize);
+        }
+      }
+    );
+
+    it.each(heightWidth)(
+      "should render Flat Table with %spx as a height parameter",
+      (height) => {
+        CypressMountWithProviders(
+          <stories.FlatTableComponent height={`${height}px`} />
+        );
+
+        flatTableWrapper().should("have.css", "height").as("wrapperHeight");
+        cy.get("@wrapperHeight")
+          .then(($el) => parseInt($el.toString()))
+          .should("be.within", height - 1, height + 1);
+      }
+    );
+
+    it.each([150, 249, 250, 251, 300])(
+      "should render Flat Table with %spx as a height parameter and minHeight set to 250px",
+      (height) => {
+        CypressMountWithProviders(
+          <stories.FlatTableComponent
+            height={`${height}px`}
+            minHeight="250px"
+          />
+        );
+
+        if (height < 250) {
+          flatTableWrapper().should("have.css", "height", `${250}px`);
+        } else {
+          flatTableWrapper().should("have.css", "height").as("wrapperHeight");
+          cy.get("@wrapperHeight")
+            .then(($el) => parseInt($el.toString()))
+            .should("be.within", height - 1, height + 1);
+        }
+      }
+    );
+
+    it("should render Flat Table with hasMaxHeight parameter", () => {
+      CypressMountWithProviders(
+        <stories.FlatTableComponent height="400px" hasMaxHeight />
+      );
+
+      flatTableWrapper().should("have.css", "max-height", "100%");
+    });
+
+    it.each(heightWidth)(
+      "should render Flat Table with %spx as a width parameter",
+      (width) => {
+        CypressMountWithProviders(
+          <stories.FlatTableComponent width={`${width}px`} />
+        );
+
+        flatTableWrapper().should("have.css", "width").as("wrapperWidth");
+        cy.get("@wrapperWidth")
+          .then(($el) => parseInt($el.toString()))
+          .should("be.within", width - 1, width + 1);
+      }
+    );
+
+    it.each(["visible", "hidden", "clip", "scroll", "auto"])(
+      "should render Flat Table with %s as a overflowX parameter and width set to 500px",
+      (overflow) => {
+        CypressMountWithProviders(
+          <stories.FlatTableComponent width="500px" overflowX={overflow} />
+        );
+
+        flatTable().parent().should("have.css", "overflow-x", overflow);
+      }
+    );
+
+    it("should render Flat Table with rowSpan set to make header cells span 2 rows", () => {
+      CypressMountWithProviders(
+        <stories.FlatTableSpanComponent width="500px" />
+      );
+
+      for (let i = 0; i < 4; i++) {
+        if (i === 0 || i === 1 || i === 3) {
```

Comment:
**suggestion:** could do the inverse here and say: 
```suggestion
        if (i !== 2 || i !== 4) {
```
uses less parameters in the `if` condition. I'm happy either way though and it's fine this way if this is how you lot in QA want the tests to be written.